File: prototypes/MessagingPrototype/src/main/scala/messaging/Employee.scala
package messaging

import javax.persistence.{Entity, GeneratedValue, Id}

/**
 * Created by justusadam on 19/11/14.
 */
@Entity
class Employee{

  @Id
  @GeneratedValue
  private var id:Int = _
  private var name:String = _
  private var sends:Boolean = _
  private var receives:Boolean = _

  def this(name:String) = {
    this()
    this.name = name
  }

  def this(id:Int, name:String, sends:Boolean, receives:Boolean) = {
    this()
    this.sends = sends
    this.receives = receives
    this.id = id
    this.name = name
  }

  def this(name:String, sends:Boolean, receives:Boolean) = {
    this()
    this.sends = sends
    this.receives = receives
    this.name = name
  }

  def getId = id

  def canSend = sends

  def canReceive = receives

  def getName = name
}

File: src/main/java/fviv/model/Employee.java
package fviv.model;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.OneToOne;

import org.salespointframework.useraccount.UserAccount;
import org.springframework.beans.factory.annotation.Autowired;

/**
 *@author Hendric Eckelt
*/

@Entity
public class Employee{
	public static enum Departement {
		MANAGEMENT, CATERING, SECURITY, CLEANING, NULL;
	}
	
	private String phone;
	private Departement departement;
	
	@Id
	@GeneratedValue
	private long id;
	
	@OneToOne
	private UserAccount userAccount;
	
	@Deprecated
	protected Employee(){}
	
	@Autowired
	public Employee(UserAccount userAccount, String lastname, String firstname, String email, String phone, Departement departement){
		this.userAccount = userAccount;
		userAccount.setLastname(lastname);
		userAccount.setFirstname(firstname);
		userAccount.setEmail(email);
		this.phone = phone;
		this.departement = departement;
	}
	
	public UserAccount getUserAccount(){
		return userAccount;
	}
		
	public String getPhone(){
		return phone;
	}
		
	public void setPhone(String phone){
		this.phone = phone;
	}
	
	public long getId(){
		return id;
	}
	
	public void setDepartement(Departement d) {
		this.departement = d;
	}
	
	public Departement getDepartement() {
		return departement;
	}
	
	public void setId(long id) {
		this.id = id;
	}
}
